src/scanner.c

```diff
@@ -79,7 +79,7 @@ static unsigned serialize(Scanner *scanner, char *buffer) {
 
     // We serialize to and from uint8_t's, we cannot currently serialize any
     // value above that.
-    uint8_t len = scanner->expected_percent_count.len;
+    uint32_t len = scanner->expected_percent_count.len;
```

Comment:
What's about the comment above? Is it obsolete? I'm not sure to understand the original meaning, tbh.

Reply:
The state serialization format just writes a ringle byte for the length of the expected percent vector, followed by the data itself. This has always been broken if someone uses more than 255 percents for a Nickel string delimiter. But it does mean that the serialization format is nice and simple.

This change just silences a compiler warning that the check `len > UINT_MAX8` is useless. If we really care about this usecase we should fix the serialization format and rewrite this part of the scanner...